sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/test/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobPartitionManagerTest.java

```diff
@@ -0,0 +1,159 @@
+        headers.put("eTag", "etag2");
+        when(containerAsyncClient.getBlobAsyncClient("eh/cg/0")).thenReturn(blobAsyncClient);
+        when(blobAsyncClient.setMetadataWithResponse(any(Metadata.class), any(BlobAccessConditions.class)))
+            .thenReturn(Mono.just(new VoidResponse(null, 200, new HttpHeaders(headers))));
+
+        BlobPartitionManager blobPartitionManager = new BlobPartitionManager(containerAsyncClient);
+        StepVerifier.create(blobPartitionManager.updateCheckpoint(checkpoint))
+            .assertNext(etag -> assertEquals("etag2", etag)).verifyComplete();
+    }
+
+
+    @Test
+    public void testClaimOwnership() {
+        PartitionOwnership po = createPartitionOwnership("eh", "cg", "0", "owner1");
+
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.put("eTag", "etag2");
+
+        when(containerAsyncClient.getBlobAsyncClient("eh/cg/0")).thenReturn(blobAsyncClient);
+        when(blobAsyncClient.asBlockBlobAsyncClient()).thenReturn(blockBlobAsyncClient);
+        when(blockBlobAsyncClient.uploadWithResponse(any(Flux.class), eq(0L),
+            isNull(), any(Metadata.class), any(BlobAccessConditions.class)))
+            .thenReturn(Mono.just(new ResponseBase(null, 200, httpHeaders, null, null)));
+
+        BlobPartitionManager blobPartitionManager = new BlobPartitionManager(containerAsyncClient);
+        StepVerifier.create(blobPartitionManager.claimOwnership(po))
+            .assertNext(partitionOwnership -> {
+                assertEquals("owner1", partitionOwnership.ownerId());
+                assertEquals("0", partitionOwnership.partitionId());
+                assertEquals("eh", partitionOwnership.eventHubName());
+                assertEquals("cg", partitionOwnership.consumerGroupName());
+                assertEquals("etag2", partitionOwnership.eTag());
+                assertNull(partitionOwnership.sequenceNumber());
+                assertNull(partitionOwnership.offset());
+            }).verifyComplete();
+    }
+
+    private PartitionOwnership createPartitionOwnership(String eventHubName, String consumerGroupName,
+        String partitionId, String ownerId) {
+        return new PartitionOwnership()
+            .eventHubName(eventHubName)
+            .consumerGroupName(consumerGroupName)
+            .partitionId(partitionId)
+            .ownerId(ownerId);
+    }
+
+    private BlobItem getBlobItem(String owner, String sequenceNumber, String offset, String etag, String blobName) {
+        Metadata metadata = getMetadata(owner, sequenceNumber, offset);
+
+        BlobProperties properties = new BlobProperties()
+            .lastModified(OffsetDateTime.now())
+            .etag(etag);
+
+        return new BlobItem()
+            .name(blobName)
+            .metadata(metadata)
+            .properties(properties);
+    }
+
+    private Metadata getMetadata(String owner, String sequenceNumber, String offset) {
+        Metadata metadata = new Metadata();
```

Comment:
Just to make sure, is metadata case sensitive? @chradek was mentioning some bug about it.. and it would be nice to interop between language libraries.

Reply:
Yeah, I discussed this with Shivangi and we will all have the same name for interoperability. I have a comment in the source class to use the same keys in all languages.

Reply:
Also, started a thread on Teams to finalize the name.